src/pages/index.js

```diff
@@ -22,96 +34,233 @@ const profileDescriptionInput = profileEditModal.querySelector(
-  user.setUserInfo(name, description);
-  editProfilePopup.close();
+// Handle image click
+function handleImageClick(card) {
+  previewImagePopup.open({ name: card.name, link: card.link });
 }
 
-function handleAddCardSubmit(newCardData, cardListElement) {
-  const name = newCardData.title;
-  const alt = newCardData.title;
-  const link = newCardData.url;
-  renderCard({ name, alt, link });
-  newCardPopup.close();
+// Handle card delete
+function handleDelete(card) {
+  confirmation.open();
+  confirmation.setConfirmSubmit(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        confirmation.close();
+        card._handleDeleteCard();
+      })
+      .catch((err) => console.log(err));
+  });
 }
 
-function createCard(cardData) {
-  return new Card(cardData, "#card-template", handleImageClick).getView();
+// Handle Like click
+function handleLikeCard(card) {
+  if (card._isLiked) {
+    api
+      .removeLike(card._id)
+      .then((res) => {
+        card.updateIsLiked(res.isLiked);
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(card._id)
+      .then((res) => {
+        card.updateIsLiked(res.isLiked);
+      })
+      .catch((err) => console.log(err));
+  }
 }
+
 /* -------------------------------------------------------------------------- */
 /*                               Event Listeners                              */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Form
+
+// Edit Profile Form
 profileEditBtn.addEventListener("click", () => {
   const userInput = user.getUserInfo();
-  editProfilePopup.setInputValues({
-    title: userInput.name,
-    subheader: userInput.about,
-  });
-  profileEditFormValidator.resetValidation();
+  profileTitleInput.value = userInput.name;
+  profileDescriptionInput.value = userInput.about;
   editProfilePopup.open();
+  profileEditFormValidator.resetValidation();
 });
 
-//New Card Form
+// New Card Form
 addCardBtn.addEventListener("click", () => {
   newCardPopup.open();
   addCardFormValidator.toggleButtonState();
 });
 
+// Delete Card Form
+const confirmation = new PopupDelete({
+  popupSelector: "#delete-card-modal",
+});
+confirmation.setEventListeners();
+
+// Avatar Edit Form
+avatarEditBtn.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
 /* -------------------------------------------------------------------------- */
-/*                               Validation                                 */
+/*                               Validation                                    */
 /* -------------------------------------------------------------------------- */
 const addCardFormValidator = new FormValidator(config, addCardForm);
 addCardFormValidator.enableValidation();
 const profileEditFormValidator = new FormValidator(config, profileEditForm);
 profileEditFormValidator.enableValidation();
+
+const avatarFormValidator = new FormValidator(config, avatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```